dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/ws/WSJobRepository.java

```diff
@@ -268,6 +269,9 @@ public abstract WSStepThreadExecutionAggregate getStepExecutionAggregateFromJobE
     public abstract WSRemotablePartitionExecution createRemotablePartition(RemotablePartitionKey remotablePartitionKey);
 
     /**
+     * Updates an entry for this remote partition in the RemotablePartition table with the given internalStatus
+     * If the remotable partition is not found, then it just returns
+     *
      * @param remotablePartitionKey
      * @return
      * @throws IllegalArgumentException if partition isn't found via key
```

Comment:
So if the above comment is right (I think it is), the `@throws` clause should be removed.

Reply:
Looks like there's a potential IllegalArgumentException on the EntityManager find method, if the entity class supplied isn't actually an entity, or if the key to look up is null or not applicable.. not sure if that realistically will ever trigger for us, but I guess it's technically possible? At the very least, the wording here is inaccurate, since not finding a match won't throw.

Reply:
Yeah..better now.. we don't always document every very-unexpected exc but should document somewhat-expected ones accurately.